src/test/java/com/google/edith/ReceiptFileHandlerServletTest.java

```diff
@@ -0,0 +1,87 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.edith;
+
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.google.appengine.api.blobstore.FileInfo;
+import com.google.appengine.tools.development.testing.LocalBlobstoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.edith.interfaces.ReceiptFileHandlerInterface;
+import com.google.edith.servlets.ReceiptFileHandlerServlet;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public final class ReceiptFileHandlerServletTest {
+  private final LocalServiceTestHelper testHelper =
+      new LocalServiceTestHelper(
+          new LocalBlobstoreServiceTestConfig(), new LocalDatastoreServiceTestConfig());
+
+  private ReceiptFileHandlerServlet receiptFileHandlerServlet;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    testHelper.setUp();
+    receiptFileHandlerServlet = new ReceiptFileHandlerServlet(receiptFileHandler);
+  }
+
+  @After
+  public void tearDown() {
+    testHelper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+
+  @Mock HttpServletResponse response;
+
+  @Mock ReceiptFileHandlerInterface receiptFileHandler;
+
+  // If no file is uploaded, Blob is not stored and Exception is thrown.
+  @Test(expected = IllegalStateException.class)
+  public void doPost_ifNoFileUpload_throwException() throws IOException {
+    List<FileInfo> files = Collections.emptyList();
+    when(receiptFileHandler.getUploadedFileUrl(request, "receipt-file")).thenReturn(files);
+
+    receiptFileHandlerServlet.doPost(request, response);
```

Comment:
Missing assertion. Can use `assertThrows` to test exceptions.

Reply:
i think ```assertThrows``` was introduced in junit 5 but we are using junit 4.12. Should we change the junit version ?